### PR TITLE
[IMP] web: convert core/network/http_service tests to hoot

### PR DESCRIPTION
**Pull request by mcm-odoo**
task-id: 3705027

### CODE REVIEW

File: addons/web/static/tests/core/network/http_service.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    expect(["GET"]).toVerifySteps();
+
+    await post("/call_post");
+    expect(["POST"]).toVerifySteps();
+});
+
+test`headless`("check status 502", async () => {
+    const restoreFetch = mockFetch(() => new Response("{}", { status: 502 }));
+    after(restoreFetch);
+    await expect(() => get("/custom_route")).rejects.toThrow();
```
Comment (aab-odoo) [inline addons/web/static/tests/core/network/http_service.test.js:23]: no specific error to expect here?

File: addons/web/static/tests/core/network/http_service.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/** @odoo-module **/
+
+import { after, expect, test } from "@odoo/hoot";
+import { mockFetch } from "@odoo/hoot-mock";
+import { get, post } from "@web/core/network/http_service";
```
Comment (aab-odoo) [inline addons/web/static/tests/core/network/http_service.test.js:5]: i would try to separate hoot imports from @web imports, as with hoot and its module set system, they come from completely different environments (hoot: once for all, others: specific to this file)

File: addons/web/static/tests/core/network/http_service.test.js (hunk 1 above)
Comment (mcm-odoo) [reply addons/web/static/tests/core/network/http_service.test.js]: yes and I've noticed this assertion passed thanks to an error thrown by hoot while adding the matcher.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**